javascript/javascript2/week2/main.js

```diff
@@ -0,0 +1,88 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector("section.products ul");
+//console.log(productsUl);
+//functions
+const resetProductUl = () => {
+	productsUl.innerHTML = "";
+};
+function renderProducts(products) {
+	resetProductUl();
+	const headingTag = document.createElement("li");
+	headingTag.innerHTML = `  <ul class="heading">
+    <li>Product Name</li>
+    <li>Price</li>
+    <li>Rating</li>
+    <li>Country</li>
+  </ul>`;
+	productsUl.appendChild(headingTag);
+	products.forEach((product) => {
+		const li = document.createElement("li");
+
+		let shipsToHTML = "";
+		product.shipsTo.forEach(
+			(country) => (shipsToHTML += `<li>${country}</li>`)
+		);
+
+		li.innerHTML = `
+            <ul>
+                <li>${product.name}</li>
+                <li>${product.price}</li>
+                <li>${product.rating}</li>
+                <ul class="ships-to">${shipsToHTML}</ul>
+            </ul>
+        `;
+		productsUl.appendChild(li);
+	});
+}
+
+//renderProducts(products);
+
+window.onload = function (event) {
+	products.sort((a, b) => b.rating - a.rating);
+	renderProducts(products);
+};
+
+const productSearchTag = document.getElementById("search-input");
+const maxPriceTag = document.getElementById("priceSearch");
+const sortOption = document.getElementById("sortSearch");
+
+productSearchTag.addEventListener("input", () => {
+	renderFilterProducts();
+});
+
+maxPriceTag.addEventListener("input", () => {
+	renderFilterProducts();
+});
+
+sortOption.addEventListener("change", () => {
+	renderFilterProducts();
+});
```

Comment:
These can be written as `htmlElement.addEventListener("change",renderFilterProduct)`

Reply:
Thank you so much for guidence. I have change code as per your suggestion.